tests/resources/SenderResource.test.ts

```diff
@@ -0,0 +1,249 @@
+import { Response } from 'node-fetch';
+import { Readable } from 'node:stream';
+
+import { AccessTokenOwnerConfig, Api, SenderResource, Requests } from '../../src';
+import { mockApiHappyPath, MockedApi, restoreMockedApi } from '../setup/mockApi';
+import { mockedFetch, mockFetchHappyPath } from '../setup/mockNodeFetch';
+
+const MOCK_TOKEN_CONFIG: AccessTokenOwnerConfig = {
+  type: 'OWNER',
+  clientId: 'clientId',
+  secret: 'secret',
+};
+const MOCK_API_URL = 'http://demo.com';
+
+describe('SenderResource', () => {
+  let mockedApi: MockedApi;
+  let senders: SenderResource;
+
+  beforeAll(() => {
+    mockFetchHappyPath();
+    mockedApi = mockApiHappyPath();
+    senders = new SenderResource(new Api(MOCK_TOKEN_CONFIG, MOCK_API_URL));
+  });
+
+  afterAll(() => {
+    restoreMockedApi(mockedApi);
+    mockedFetch.mockReset();
+  });
+
+  afterEach(() => {
+    mockedFetch.mockClear();
+  });
+
+  describe('create', () => {
+    it('sends the provided payload as request body', async () => {
+      const payload: Requests.CreateSenderData = {
+        name: 'sender',
+        firstName: 'sender first name',
+        lastName: 'sender last name',
+      };
+
+      await senders.create(payload);
+
+      expect(mockedFetch.mock.calls[0]).toMatchInlineSnapshot(`
```

Comment:
Nice choice of using inline snapshots! I guess if the snapshot ends up taking a screen's total height we might want to consider switching to using the regular `toMatchSnapshot`.

Reply:
Good point. I think i'll try to do the same going forward on smaller code blocks. Having to go to another file sometimes means never going there to check whereas if it's available in your face there, you verify it.